pytorch_lightning/plugins/training_type/ddp_comm_hook_util.py

```diff
@@ -0,0 +1,57 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+from typing import Optional
+
+from pytorch_lightning.utilities import (
+    _TORCH_GREATER_EQUAL_1_9,
+    rank_zero_warn,
+    rank_zero_info,
+)
+from torch.nn.parallel.distributed import DistributedDataParallel
+
+
+def register_ddp_comm_hook(
+    ddp_comm_state: Optional[object],
+    ddp_comm_hook: Optional[callable],
+    ddp_comm_wrapper: Optional[callable],
+    model: DistributedDataParallel,
+    is_single_process_single_device: bool,
+) -> None:
```

Comment:
please add a docstring along with an example usage

Reply:
may I suggest to put the model argument as the first argument? It would be more consistent with the rest of lightning api where model usually comes first.